src/credentials/PersistentStorageOpCertStore.cpp

```diff
@@ -494,5 +712,59 @@ CHIP_ERROR PersistentStorageOpCertStore::GetCertificate(FabricIndex fabricIndex,
     return LoadCertFromStorage(mStorage, fabricIndex, element, outCertificate);
 }
 
+CHIP_ERROR PersistentStorageOpCertStore::GetVidVerificationElement(FabricIndex fabricIndex, VidVerificationElement element,
+                                                                   MutableByteSpan & outElement) const
+{
+    VerifyOrReturnError(mStorage != nullptr, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex), CHIP_ERROR_INVALID_FABRIC_INDEX);
+
+    StorageKeyName keyName = StorageKeyName::FromConst("");
+    if (element == VidVerificationElement::kVidVerificationStatement)
+    {
+        if (mStateFlags.Has(StateFlags::kVidVerificationStatementUpdated) && (fabricIndex == mPendingFabricIndex))
+        {
+            return CopySpanToMutableSpan(
+                ByteSpan{ mPendingVidVerificationStatement.Get(), mPendingVidVerificationStatement.AllocatedSize() }, outElement);
+        }
+
+        keyName = DefaultStorageKeyAllocator::FabricVidVerificationStatement(fabricIndex);
+    }
+
+    if (element == VidVerificationElement::kVvsc)
+    {
+        if (mStateFlags.Has(StateFlags::kVvscUpdated) && (fabricIndex == mPendingFabricIndex))
+        {
+            return CopySpanToMutableSpan(ByteSpan{ mPendingVvsc.Get(), mPendingVvsc.AllocatedSize() }, outElement);
+        }
+
+        keyName = DefaultStorageKeyAllocator::FabricVVSC(fabricIndex);
+    }
+
+    if (!keyName)
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    uint8_t storageBuffer[kMaxCHIPCertLength];
+    uint16_t keySize = sizeof(storageBuffer);
+    static_assert(kMaxCHIPCertLength > (2 * (Crypto::kVendorIdVerificationStatementV1Size)),
```

Comment:
Does this have alternate SDK impact as well? ie, should this be something that's specifically called out in the spec?

Reply:
It doesn't really. This is cheap A-B-A issues prevention attempt.